File: FestivalJava/Model/src/main/java/Domain/Show.java
package Domain;

import java.io.Serializable;
import java.util.Date;

/**
 * Created by Sebi on 09-Mar-17.
 */
public class Show implements Serializable{
    //id
    private Integer idShow;

    //location of the show
    private String location;

    //date and time
    private Date date;

    //how many tickets are available
    private Integer ticketsAvailable;

    //how many tickets have been sold
    private Integer ticketsSold;

    //artist taking part in the show
    private Artist artist;

    //Constructor
    public Show(Integer idShow, String location, Date date, Integer ticketsAvailable, Integer ticketsSold, Artist artist){
        this.idShow = idShow;
        this.location = location;
        this.date = date;
        this.ticketsAvailable = ticketsAvailable;
        this.ticketsSold = ticketsSold;
        this.artist = artist;
    }

    //getters and setters
    public Integer getIdShow() {
        return idShow;
    }

    public void setIdShow(Integer idShow) {
        this.idShow = idShow;
    }

    public String getLocation() {
        return location;
    }

    public void setLocation(String location) {
        this.location = location;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public Integer getTicketsAvailable() {
        return ticketsAvailable;
    }

    public void setTicketsAvailable(Integer ticketsAvailable) {
        this.ticketsAvailable = ticketsAvailable;
    }

    public Integer getTicketsSold() {
        return ticketsSold;
    }

    public void setTicketsSold(Integer ticketsSold) {
        this.ticketsSold = ticketsSold;
    }


    public Artist getArtist() {
        return artist;
    }

    public void setArtist(Artist artist) {
        this.artist = artist;
    }

    @Override
    public String toString(){
        return idShow + " " + location + " " + date + " " +
                ticketsAvailable + " " + ticketsSold + " " + artist.getName();
    }

    @Override
    public boolean equals(Object obj){
        if (obj == null){
            return false;
        }
        if (!(obj instanceof Show)){
            return false;
        }
        Show show = (Show)obj;
        return this.idShow == show.idShow;
    }
}

File: FestivalC#/Festival/Model/Model/Show.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Model
{
    public class Show
    {
        //id
        private int idShow;

        //location of the show
        private String location;

        //date 
        private DateTime date;

        //number of available tickets
        private int ticketsAvailable;

        //number of sold tickets
        private int ticketsSold;

        //id of the artist
        private Artist artist;

        //constructor
        public Show(int idShow, String location, DateTime date, int ticketsAvailable, int ticketsSold, Artist artist)
        {
            this.idShow = idShow;
            this.location = location;
            this.date = date;
            this.ticketsAvailable = ticketsAvailable;
            this.ticketsSold = ticketsSold;
            this.artist = artist;
        }

        public int IdShow
        {
            get
            {
                return idShow;
            }

            set
            {
                idShow = value;
            }
        }

        public string Location
        {
            get
            {
                return location;
            }

            set
            {
                location = value;
            }
        }

        public DateTime Date
        {
            get
            {
                return date;
            }

            set
            {
                date = value;
            }
        }

        public int TicketsAvailable
        {
            get
            {
                return ticketsAvailable;
            }

            set
            {
                ticketsAvailable = value;
            }
        }

        public int TicketsSold
        {
            get
            {
                return ticketsSold;
            }

            set
            {
                ticketsSold = value;
            }
        }

        public Artist Artist
        {
            get
            {
                return artist;
            }

            set
            {
                artist = value;
            }
        }

        public override bool Equals(object obj)
        {
            if (obj == null)
            {
                return false;
            }
            if (!(obj is Show))
            {
                return false;
            }
            Show s = (Show)obj;
            return s.idShow == idShow && s.location.Equals(location)
                && s.date.Equals(date) && s.ticketsAvailable == ticketsAvailable 
                && s.ticketsSold == ticketsSold && s.artist.Equals(artist);
        }

        public override string ToString()
        {
            return idShow + " " + location + " " + date.ToString() + " " + ticketsAvailable + " " + ticketsSold + " " + artist.IdArtist;
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }
    }
}
